sky/jobs/controller.py

```diff
@@ -291,13 +313,29 @@ def _run_one_task(self, task_id: int, task: 'sky.Task') -> bool:
                     continue
                 elif job_status in job_lib.JobStatus.user_code_failure_states():
                     # The user code has probably crashed, fail immediately.
-                    end_time = managed_job_utils.get_job_timestamp(
-                        self._backend, cluster_name, get_end_time=True)
+                    try:
+                        end_time = managed_job_utils.get_job_timestamp(
+                            self._backend, cluster_name, get_end_time=True)
+                    except exceptions.CommandError as e:
+                        if e.returncode == 255:
+                            end_time = time.time()
+                            logger.info(
+                                f'Failed to connect to cluster {cluster_name} '
+                                'to fetch job end time. Assuming the cluster '
+                                'was preempted, and using the current time.')
+                        else:
+                            raise
                     logger.info(
                         'The user job failed. Please check the logs below.\n'
                         f'== Logs of the user job (ID: {self._job_id}) ==\n')
 
-                    self._download_log_and_stream(task_id, handle)
+                    try:
+                        self._download_log_and_stream(task_id, handle)
```

Comment:
Seem we already have the error handling in the function. Should we handle it again here?

Reply:
The handling inside the function only handles some case. I think we should catch a broader exception here. I will check if we can move the handling inside the function without breaking skyserve usage.